crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -60,8 +63,8 @@ ruma_api! {
         pub unsubscribe_rooms: &'a [OwnedRoomId],
 
         /// Extensions API.
-        #[serde(skip_serializing_if = "BTreeMap::is_empty")]
-        pub extensions: BTreeMap<String, serde_json::Value>,
+        #[serde(skip_serializing_if = "ExtensionsRequest::is_empty")]
```

Comment:
`default` and `skip_serializing_if` must always be matched, except on `Option` where `default` is implied.

```suggestion
        #[serde(default, skip_serializing_if = "ExtensionsRequest::is_empty")]
```